client/components/infinite-scroll/index.js

```diff
@@ -0,0 +1,57 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { defer, throttle } from 'lodash';
+
+const SCROLL_CHECK_RATE_IN_MS = 400;
+
+export default class InfiniteScroll extends Component {
+	static propTypes = {
+		nextPageMethod: PropTypes.func.isRequired,
+	};
+
+	checkScrollPositionHandler = throttle(
+		this.checkScrollPosition.bind( this, true ),
+		SCROLL_CHECK_RATE_IN_MS
+	);
+
+	componentDidMount() {
+		window.addEventListener( 'scroll', this.checkScrollPositionHandler );
+		this.checkScrollPosition( false );
+	}
+
+	componentDidUpdate() {
+		this.checkScrollPosition( false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'scroll', this.checkScrollPositionHandler );
+	}
+
+	checkScrollPosition( triggeredByScroll ) {
+		const scrollPosition = window.pageYOffset;
+		const documentHeight = document.body.scrollHeight;
+		const viewportHeight = window.innerHeight;
+		const scrollOffset = 2 * viewportHeight;
+
+		if ( scrollPosition >= documentHeight - viewportHeight - scrollOffset ) {
+			// Consider all page fetches once user starts scrolling as triggered by scroll
+			// Same condition check is in components/infinite-list/scroll-helper loadNextPage
+			if ( scrollPosition > viewportHeight ) {
+				triggeredByScroll = true;
+			}
+
+			// scroll check may be triggered while dispatching an action,
+			// we cannot create new action while dispatching old one
+			defer( () => this.props.nextPageMethod( { triggeredByScroll } ) );
```

Comment:
this was kind of a flux-ism. I wonder if we can ditch it... Having it in there is a little weird.

Reply:
I️ thought the same thing but then remembered that the fewer sync behaviors that occur in lifecycle methods the better

Reply:
Good point, I'm removing the `defer` call. It shouldn't be needed in Redux. Leaving it there "just in case" would be just superstition -- we should have known and specific reasons to add mysterious `defer` calls like this.